File: Programmers/주식가격.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;
//새로 나온것이 기존것보다 작으면 무언가를 해주어야한다.
//스택에 쌓다가 작은게 나오면
// 숫자가 구분이 안되니까 인덱스로 푸쉬해준다.
vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size());
    stack <int> st;
    for(int i = 0 ; i < prices.size() ; i++){
        while(!st.empty() && prices[st.top()] > prices[i]){
            answer[st.top()] = i - st.top();
            st.pop();
        }
        st.push(i);
    }
        while(!st.empty()){
            answer[st.top()] = prices.size() - st.top() - 1;
            st.pop();
        }
    return answer;
}

File: CNS_java/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        int cnt = 0;
        for(int i = 0 ; i < prices.length ; i++){
            for(int j = i+1 ; j < prices.length ; j++){
                cnt++;
                if(prices[i] > prices[j]) break;
            }
            answer[i] = cnt;
            cnt = 0;
        }
        return answer;
    }
}
